modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/fragment/renderer/FDSAdminFragmentRenderer.java

```diff
@@ -939,11 +937,12 @@ private JSONArray _getSelectedItemsJSONArray(
 			preselectedValues);
 
 		for (int i = 0; i < preselectedValuesJSONArray.length(); i++) {
-			String key = preselectedValuesJSONArray.getString(i);
+			JSONObject jsonObject = preselectedValuesJSONArray.getJSONObject(i);
 
 			for (ListTypeEntry listTypeEntry : listTypeEntries) {
 				if (Objects.equals(
-						listTypeEntry.getExternalReferenceCode(), key)) {
+						listTypeEntry.getExternalReferenceCode(),
+						jsonObject.getString("value"))) {
```

Comment:
`preselectedValues` db field / state variable had two formats, for two source types. One was an array of label/value objects, the other array of strings (ERCs). This complicated code. Now it's always objects. We are sacrificing a tiny bit of performance, as we have a slightly bigger footprint in DB that necessary, but I think it's worth it for the code simplification.